src/base/preferences.cpp

```diff
@@ -718,6 +718,16 @@ void Preferences::useSystemIconTheme(bool enabled)
 }
 #endif
 
+bool Preferences::useSystemFileIcons() const
```

Comment:
Since you're still not keeping track of its changes, maybe just use SettingsStorage directly?
Or create appropriate Application property (since we have no separate GUI component class).